metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -308,6 +308,21 @@ def construct_chart_from_chart_data(self, chart_data):
         group_bys = params.get("groupby", []) or []
         if isinstance(group_bys, str):
             group_bys = [group_bys]
+        # handling List[Union[str, dict]] case
+        # a dict containing two keys: sqlExpression and label
+        elif isinstance(group_bys, list) and len(group_bys) != 0:
+            temp_group_bys = []
+            for item in group_bys:
+                # if the item is a custom label
+                if isinstance(item, dict):
+                    item_value = item.get("label", "")
+                    if item_value != "": temp_group_bys.append(f"{item_value}_custom_label")
```

Comment:
Why do we need _custom_label here?

Reply:
The "_custom_label" acts as an identifier since the original label is lost when using a custom label. As mentioned in the description above, a future improvement is to somehow retrieve the lost value, but that's up to the Superset API.